libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1312,9 +1312,13 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 		}
 		else
 		{
-			IRVariable argument = convert(*arguments.front(), *arrayType.baseType());
+			IRVariable argument =
+				arrayType.baseType()->isValueType() ?
+convert(*arguments.front(), *arrayType.baseType()) :
```

Comment:
Ah sorry, my suggestions had wrong tabs, so this went wrong. There might be others too.

Reply:
I think others were good, but will check. I fixed this one now